core-processor/src/main/java/io/micronaut/inject/beans/visitor/EntityIntrospectedAnnotationMapper.java

```diff
@@ -20,7 +20,7 @@
 import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.ReflectiveAccess;
-import io.micronaut.inject.annotation.NamedAnnotationMapper;
+import io.micronaut.inject.annotation.processor.NamedAnnotationMapper;
```

Comment:
we can't change the package of this interface it would be a major breaking change to the framework and downstream modules

Reply:
The only other solution that i saw was a user facing breaking change in the sense that they need to add a new annotation-processor dependency to make `:context` work.

Yes this is a breaking change for projects such as `micronaut-openapi` or `micronaut-data` but i believe this is the best and most future-proof solution as described in the related commit.

Shouldn't it be possible to adapt all "downstream projects" and ship the new version with the BOM. The only thing that would change as far as i can see is that `micronaut-data:3.x.x` is not compatible with `micronaut 4.x.x`, which i don't see a big problem with. Yes it would be better if all 2.x  & 3.x modules still worked with 4.x core but i would deem this to be acceptable to have an internal breaking change that is not user-facing

Reply:
@graemerocher hi, I wanted to ask back if this is a hard "no" criteria or do you see any other solutions to this problem.

to summarize the problem in short words: the component `:context` should be a module. `:context` depends (compile time dependencies are dependencies too) on two modules `:inject` and `:core-processor` which both expose the same package `io.micronaut.inject.annotation`.  2 dependencies exposing the same named package to a JPMS module is not possible.